### ISSUE TITLE
panic with peer / addrmgr

### ISSUE THREAD

**Issue by dajohi** [btcsuite/btcd#579]
```text
panic: runtime error: invalid memory address or nil pointer dereference
[signal 0xb code=0x1 addr=0x38 pc=0x6876a8]

goroutine 827 [running]:
github.com/btcsuite/btcd/addrmgr.NetAddressKey(0x0, 0x0, 0x0)
        /home/dhill/.go/src/github.com/btcsuite/btcd/addrmgr/addrmanager.go:734 +0x28
main.(*serverPeer).OnVersion(0xc8200a8400, 0xc82220c840, 0xc82200da00)
        /home/dhill/.go/src/github.com/btcsuite/btcd/server.go:345 +0x3f7
main.(*serverPeer).OnVersion-fm(0xc82220c840, 0xc82200da00)
        /home/dhill/.go/src/github.com/btcsuite/btcd/server.go:1405 +0x34
github.com/btcsuite/btcd/peer.(*Peer).inHandler(0xc82220c840)
        /home/dhill/.go/src/github.com/btcsuite/btcd/peer/peer.go:1532 +0xe62
created by github.com/btcsuite/btcd/peer.(*Peer).Start
        /home/dhill/.go/src/github.com/btcsuite/btcd/peer/peer.go:2017 +0x2c9

goroutine 1 [chan receive, 7 minutes]:
main.btcdMain(0x0, 0x0, 0x0)
        /home/dhill/.go/src/github.com/btcsuite/btcd/btcd.go:133 +0xe5d
main.main()
        /home/dhill/.go/src/github.com/btcsuite/btcd/btcd.go:163 +0x184
```

**Comment by dajohi** [comment]
Perhaps @tuxcanfly can take a look...

**Comment by dajohi** [comment]
I believe handleVersionMsg in peer needs to return an error, then
```golang
 case *wire.MsgVersion:
                        p.handleVersionMsg(msg)
                        if p.cfg.Listeners.OnVersion != nil {
                                p.cfg.Listeners.OnVersion(p, msg)
                        }
```
needs to not continue to the listener.  Otherwise, the server gets a peer without NA set, which results in a nil dereference.

**Comment by tuxcanfly** [comment]
Thanks @dajohi I've pushed a fix based on the suggestions at #580
